nep-25.mediawiki

```diff
@@ -0,0 +1,362 @@
+
+<code>parameters</code> is an array of Parameter objects which describe the details of each parameter in the method.
+
+<code>returntype</code> indicates the return type of the method. It can have any value of the ParameterType enumeration defined below.
+
+<code>extendedreturntype</code> is an ExtendedType object used to provide additional type data if needed, this field can be omitted.
+
+===Event===
+
+Event object has the following structure:
+
+<pre>
+{
+  "name": "refund",
+  "parameters": []
+}
+</pre>
+
+<code>name</code> is the name of the event, which can be any valid identifier.
+
+<code>parameters</code> is an array of Parameter objects which describe the details of each parameter in the event.
+
+===Parameter===
+
+Parameter object has the following structure:
+
+<pre>
+{
+  "name": "from",
+  "type": "Hash160",
+  "extendedtype": {}
+}
+</pre>
+
+<code>name</code> is the name of the parameter, which can be any valid identifier.
+
+<code>type</code> indicates the type of the parameter. It can have any value of the ParameterType enumeration defined below, except for <code>Void</code>.
+
+<code>extendedtype</code> is an ExtendedType object used to provide additional type data if needed, this field can be omitted.
+
+===ParameterType===
+
+ParameterType enum has the following values:
+
+{|
+!name
+!description
+|-
+| Signature
+| A signature of a transaction or block which is generated by the user.
+|-
+| Boolean
+| A boolean value can be either <code>true</code> or <code>false</code>.
+|-
+| Integer
+| An arbitrarily large integer whose value in theory has no upper or lower bounds.
+|-
+| Hash160
+| A 160-bits integer.
+|-
+| Hash256
+| A 256-bits integer.
+|-
+| ByteArray
+| A byte array.
+|-
+| PublicKey
+| An ECC public key which is encoded with compressed mode.
+|-
+| String
+| A string which is encoded in UTF-8.
+|-
+| Array
+| An array of objects. The type of elements can be any value of ParameterType, unless some more specific type is specified with ExtendedType data.
+|-
+| Map
+| A map of objects. The type of elements inside the key/value collection can be any value of ParameterType, unless some more specific types are specified with ExtendedType data.
+|-
+| InteropInterface
+| An interface which is returned by interop services. More specific type can be specified with ExtendedType data.
+|-
+| Any
+| Any means that the method will return a value of uncertain type.
+|-
+| Void
+| Void means that the method has no return value. This value cannot be the type of a method parameter.
+|}
+
+===ExtendedType===
+
+Extended type data is an object can be used for <code>extendedtype</code>
+and <code>extendedreturntype</code> fields or named types
+(the object below is not a valid ExtendedType specification, rather it
+represents all possible fields).
+
+<pre>
+{
+  "type": "Integer",
+  "namedtype": "name",
```

Comment:
Add `AllowNull: boolean` ?

Reply:
Made it the other way around with default to allow null and a field to forbid them. I think it's safer as a default given the behavior of NEP-14, but this can be discussed.